src/uri/validator/urivalidator.rs

```diff
@@ -186,19 +185,218 @@ impl UriValidator {
+    ///         id: Some(29999),
+    ///         version_major: Some(254),
+    ///         ..Default::default()
+    ///     })
+    ///     .into(),
+    ///     resource: Some(UResource {
+    ///         id: Some(29999),
+    ///         ..Default::default()
+    ///     })
+    ///     .into(),
+    ///     ..Default::default()
+    /// };
+    /// # assert_eq!(uri.clone().authority.unwrap().id.unwrap(), expected_id_bytes);
+    /// let val_micro_form = UriValidator::validate_micro_form(&uri);
+    /// assert!(val_micro_form.is_err());
+    /// ```
+    ///
+    /// ## Overflowing UEntity ID's 16 bit capacity
+    /// ```
+    /// use uprotocol_sdk::uprotocol::{UUri, UEntity, UResource};
+    /// use uprotocol_sdk::uri::validator::{UriValidator, ValidationError};
+    ///
+    /// let uri = UUri {
+    ///     entity: Some(UEntity {
+    ///         id: Some(0x10000), // <- exceeds allotted 16 bits
+    ///         version_major: Some(254),
+    ///         ..Default::default()
+    ///     })
+    ///     .into(),
+    ///     resource: Some(UResource {
+    ///         id: Some(29999),
+    ///         ..Default::default()
+    ///     })
+    ///     .into(),
+    ///     ..Default::default()
+    /// };
+    /// let val_micro_form = UriValidator::validate_micro_form(&uri);
+    /// assert!(val_micro_form.is_err());
+    /// ```
+    ///
+    /// ## Overflowing UEntity Major Version 8 bit capacity
+    /// ```
+    /// use uprotocol_sdk::uprotocol::{UUri, UEntity, UResource};
+    /// use uprotocol_sdk::uri::validator::{UriValidator, ValidationError};
+    ///
+    /// let uri = UUri {
+    ///     entity: Some(UEntity {
+    ///         id: Some(29999),
+    ///         version_major: Some(0x100), // <- exceeds allotted 8 bits
+    ///         ..Default::default()
+    ///     })
+    ///     .into(),
+    ///     resource: Some(UResource {
+    ///         id: Some(29999),
+    ///         ..Default::default()
+    ///     })
+    ///     .into(),
+    ///     ..Default::default()
+    /// };
+    /// let val_micro_form = UriValidator::validate_micro_form(&uri);
+    /// assert!(val_micro_form.is_err());
+    /// ```
+    ///
+    /// ## Overflowing UResource ID's 16 bit capacity
+    /// ```
+    /// use uprotocol_sdk::uprotocol::{UUri, UEntity, UResource};
+    /// use uprotocol_sdk::uri::validator::{UriValidator, ValidationError};
+    ///
+    /// let uri = UUri {
+    ///     entity: Some(UEntity {
+    ///         id: Some(29999),
+    ///         version_major: Some(254),
+    ///         ..Default::default()
+    ///     })
+    ///     .into(),
+    ///     resource: Some(UResource {
+    ///         id: Some(0x10000), // <- exceeds allotted 16 bits
+    ///         ..Default::default()
+    ///     })
+    ///     .into(),
+    ///     ..Default::default()
+    /// };
+    /// let val_micro_form = UriValidator::validate_micro_form(&uri);
+    /// assert!(val_micro_form.is_err());
+    /// ```
```

Comment:
is there also an example showing a successful validation? Maybe as the first example?

Reply:
Yeah, I could add a successful validation :+1: